### ISSUE TITLE
feat: using module workers instead of traditional workers

### ISSUE THREAD

**Issue by vinayakkulkarni** [mapbox/mapbox-gl-js#9141]
## Motivation

Drastically improving the performance on slower devices using moduled workers. This is still very new and currently only in Chrome 80, but will become industry standard soon.

### Concepts & Implementation

https://web.dev/module-workers/#enter-module-workers

**Comment by mourner** [comment]
Yeah, this could be interesting to explore when browser support gets a little better — we would have to provide a different type of bundle for those.

> Drastically improving the performance on slower devices using moduled workers.

Can you expand more on the performance improvements? As far as I understand, the only major one would be browsers having to parse shared code only once (instead of each time a worker is instantiated), which could improve time-to-first-render. Are there other performance advantages?
